cdc/cdc_options.hh

```diff
@@ -27,15 +27,24 @@
 
 namespace cdc {
 
-enum class delta_mode {
+enum class delta_mode : uint8_t {
     off,
     keys,
     full,
 };
 
+enum class image_mode : uint8_t {
+    off, 
+    on,
+    full,
+};
+
+std::ostream& operator<<(std::ostream& os, delta_mode);
+std::ostream& operator<<(std::ostream& os, image_mode);
+
 class options final {
     bool _enabled = false;
-    bool _preimage = false;
+    image_mode _preimage = image_mode::off;
     bool _postimage = false;
     delta_mode _delta_mode = delta_mode::full;
```

Comment:
@haaawk @elcallio @kbr- a while back, you talked about the "delta_mode" option not being called "delta" (like I thought it should be called) because it's not a on/off but a an enum. So in this patch, "preimage" also becomes an enum. I wonder if the naming shouldn't be more consistent - i.e., delta/preimage or delta_mode/preimage_mode?

Reply:
I called it `image_mode` because eventually it could maybe be applied to post image as well, i.e. allow having non-full post image. Technically, delta_mode could also use image_mode, but I did not make this change (yet).

Reply:
Sorry, delta_mode cannot use it of course, because `keys`... doh.

Reply:
I'll commit your patch with these names, but I ask you (and @haaawk) to take another look at this issue later. My comment not about the name of the type - it's about the name of the variable and user-facing configuration, which previously you called "delta_mode" because you said "delta" would have implied true/false only. But now also "preimage" can be more than true/false, so why can't we call the delta option just "delta" instead of "delta_mode"? It's not a disaster, but it's just inconsistent.